cci_nepal/analysis/Data Analysis of Hill Data.py

```diff
@@ -0,0 +1,653 @@
+    **{"max-width": "80px", "font-size": "10pt"}
+).set_caption("Hover to magify").set_precision(2).set_table_styles(magnify())
+
+
+# All clothing related items are highly correlated in terms of preferences, maybe exhibiting same grouping in terms of respondents' perception towards them.
+
+# Interetingly, sacking bag and nylon rope also have relatively high correlation.
+
+# In[38]:
+
+
+df_nfri_non_basic = df_clean.iloc[:, 50:61]
+nfri_non_basic_corr = df_nfri_non_basic.corr()
+
+cmap = sns.diverging_palette(5, 250, as_cmap=True)
+
+nfri_non_basic_corr.style.background_gradient(cmap, axis=1).set_properties(
+    **{"max-width": "80px", "font-size": "10pt"}
+).set_caption("Hover to magify").set_precision(2).set_table_styles(magnify())
+
+
+# Here also, we can see the high correlation score between items that exhibt similar grouping. Like Laundry Soap and Bathing Soap being very highly correlated, and Tooth Brush and Paste also being highly correlated to both.
+
+# Similarly, Ladies Underwear and Sanitary Pad are also highly correlated.
+
+# #### Multi Variable Analysis ####
+
+# In[39]:
+
+
+df_clean.groupby("What.is.the.Ethnicity.of.your.household.")[
+    ["How.many.members.of.your.house.contribute.to.the.household.income."]
+].mean().sort_values(
+    ["How.many.members.of.your.house.contribute.to.the.household.income."],
+    ascending=False,
+).head()
+
+
+# In[40]:
+
+
+df_variables = df_clean.iloc[:, [2, 3, 4, 5, 6, 37]]
+variables_corr = df_variables.corr()
+
+cmap = sns.diverging_palette(5, 250, as_cmap=True)
+
+variables_corr.style.background_gradient(cmap, axis=1).set_properties(
+    **{"max-width": "80px", "font-size": "10pt"}
+).set_caption("Hover to magify").set_precision(2).set_table_styles(magnify())
+
+
+# #### Gender Wise Exploration ####
+
+# #### First, diving into respondent gender ####
+
+# In[41]:
+
+
+df_gender_respondent_grouped = (
+    df_clean.loc[df_clean["What.is.the.Gender.of.the.respondent."] != "Other"]
+    .groupby(
+        "What.is.the.Gender.of.the.respondent.",
+        as_index=False,
+    )
+    .apply(np.mean)
+)
+df_gender_respondent_grouped.loc[:, nlf.nfri_basic]
+
+# Female is 0 and Male is 1 below.
+
+
+# In[42]:
+
+
+df_gender_respondent_grouped = (
+    df_clean.loc[df_clean["What.is.the.Gender.of.the.respondent."] != "Other"]
+    .groupby(
+        "What.is.the.Gender.of.the.respondent.",
+        as_index=False,
+    )
+    .apply(np.mean)
+)
+df_gender_respondent_grouped.loc[:, nlf.nfri_non_basic]
+
+# Female is 0 and Male is 1 below.
+
+
+# Across all the items (both basic and non-basic), the importance scores are almost consistently same across both the gender respondents. Gender specific items are slighly rated higher by the respective gender, which is very understandable too.
+
+# #### Second, diving into gender ratio (Male/Female) ####
+
+# In[43]:
+
+
+df_clean["Gender.Ratio"] = df_clean["Total_Male"] / df_clean["Total_Members"]
```

Comment:
I think these sorts of features will be good for the model as will hopefully give a good indication of what items the hh will want. Lets see when it comes to feature engineering but we might want to think about an age + gender split like: majority_male_children or female_single_adult or female_main_respondent (will depend on the size of these categories if they are significant)